pyfr/backends/base/makoutil.py

```diff
@@ -14,6 +15,11 @@ def ndrange(context, *args):
     return util.ndrange(*args)
 
 
+def ilog2range(context, x):
+    n = int(math.log(x, 2))
+    return [x//(2**(i + 1)) for i in range(n)]
```

Comment:
I still do not like this.  Try using ``x.bit_length()`` as an alternative to log.